src/hott/06-contractible.rzk.md

```diff
@@ -407,27 +407,48 @@ separate hypothesis.
 #def WeakFunExt : U
   :=
     ( A : U ) → (C : A → U) →
+    -- TODO: rename f to is-contr-C (also below)?
     (f : (a : A) → is-contr (C a) ) →
     (is-contr ( (a : A) → C a ))
 
 ```
 
-For future reference we add a variable we can assume.
+Weak function extensionality implies function extensionality.
+
 
 ```rzk
-#assume weakfunext : WeakFunExt
-```
+-- TODO: delete
+#assume hole : (A : U) → A
 
-Whenever a definition (implicitly) uses function extensionality, we write
-`#!rzk uses (weakfunext)`.
+#def funext-weakfunext
+  ( weakfunext : WeakFunExt )
+  : FunExt
+  := hole FunExt
+```
 
 ```rzk
-#def call-weakfunext uses (weakfunext)
-  ( A : U )
-  ( C : A → U)
-  ( f : (a : A) → is-contr (C a) )
-  : (is-contr ( (a : A) → C a ))
-  := weakfunext A C f
+-- TODO: better name? or inline this?
+#def map-weakfunext-funext
+  (A : U)
+  (C : A → U)
+  (f : (a : A) → is-contr (C a))
+  : (a : A) → C a
+  :=
+  \ a → first (f a)
+
+#def weakfunext-funext
+  (funext : FunExt)
+  : WeakFunExt
+  :=
+  ( \ A → \ C → \ f →
```

Comment:
Here you don't need the opening parentheses (after the :=) and can collect all the lambdas as `\ A C f →`